code/modules/client/preferences/migrations/tts_blip_migration.dm

```diff
@@ -0,0 +1,12 @@
+/// Previously, tts enabled/blip were individual buttons
+/// PR #76558 changed them to one dropdown choice.
+/// This migration transfers the player's existing preferences into the new dropdown
+
+/datum/preferences/proc/update_tts_blip_prefs()
+	var/sound_blips_enabled = savefile.get_entry("sound_tts_blips")
+	if(sound_blips_enabled)
+		write_preference(GLOB.preference_entries[/datum/preference/choiced/sound_tts], TTS_SOUND_BLIPS)
+		return
+	var/tts_enabled = savefile.get_entry("sound_tts")
+	if(!tts_enabled)
+		write_preference(GLOB.preference_entries[/datum/preference/choiced/sound_tts], TTS_SOUND_OFF)
```

Comment:
Can we avoid calling `write_preference` here? It will break if we change what sound_tts is again.

Reply:
What alternatives do we have here?

Reply:
Writing to the save file key directly

Reply:
write preference writes to the private var ``tree`` which is what holds all of this, I don't think we can edit that directly without the write_preference helper or making the var not private, which I think will set bad precedent.